src/Events/QR/QR_Code.php

```diff
@@ -200,15 +219,31 @@ private function delete_qr_image( int $post_id, int $size ): void {
 	 */
 	public function render_modal(): void {
 		$post = get_post( tec_get_request_var( 'post_id' ) );
-		if ( ! $post || ! tribe_is_event( $post ) ) {
-			wp_die( esc_html__( 'Invalid event.', 'the-events-calendar' ) );
+		if ( ! $post ) {
+			wp_die( esc_html__( 'No post found.', 'the-events-calendar' ) );
+		}
+
+		if ( has_action( 'tribe_common_loaded', 'tribe_register_pro' ) ) {
+			if ( ! tribe_is_event( $post ) && ! tribe_is_event_series( $post ) ) {
+				wp_die( esc_html__( 'Not an Event or Series.', 'the-events-calendar' ) );
```

Comment:
Don't reference series in TEC.
If you have text that needs to do so, put it in ECP and leverage hooks to inject it.